stablehlo/dialect/TypeInference.h

```diff
@@ -219,10 +262,63 @@ LogicalResult inferTriangularSolveOp(
 LogicalResult inferWhileOp(Optional<Location> location, ValueRange operand,
                            SmallVectorImpl<Type>& inferredReturnTypes);
 
+//===----------------------------------------------------------------------===//
+// Verifiers for ops.
+//===----------------------------------------------------------------------===//
+
+LogicalResult verifyAllReduceOp(Optional<Location> location, Value operand,
+                                DenseIntElementsAttr replicaGroups,
+                                Region& computation);
+
+LogicalResult verifyBitcastConvertOp(Optional<Location> location, Value operand,
+                                     Value result);
+
+LogicalResult verifyBroadcastOp(Optional<Location> location,
+                                DenseIntElementsAttr sizes);
```

Comment:
Let's merge this function into `inferBroadcastOp`? Same feedback for `verifyClampOp`, `verifyDynamicSliceOp`, `verifyGetTupleElementOp` and `verifyTupleOp`.